File: app/src/main/java/com/unho/unhomeals/LoginActivity.java
package com.unho.unhomeals;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Toast;

import com.journeyapps.barcodescanner.CaptureManager;
import com.journeyapps.barcodescanner.DecoratedBarcodeView;
import com.unho.unhomeals.data.SharedPf;

public class LoginActivity extends AppCompatActivity {

    CaptureManager manager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        DecoratedBarcodeView dbv_barcode = findViewById(R.id.dbv_scanner);

        manager = new CaptureManager(this, dbv_barcode);
        manager.initializeFromIntent(getIntent(), savedInstanceState);
        manager.decode();

        dbv_barcode.decodeContinuous(result -> {
            String UID = result.toString();
            if (UID.charAt(0) == 'S') {
                SharedPf.put(this, "UID", UID);
                Intent intent = new Intent(this, MainActivity.class);
                intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK | Intent.FLAG_ACTIVITY_NEW_TASK);
                startActivity(intent);
            } else {
                String message = "잘못된 바코드입니다. 다시 시도해주세요.";
                Toast.makeText(getApplicationContext(), message, Toast.LENGTH_SHORT).show();
                finish();
            }
        });
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        manager.onDestroy();
    }

    @Override
    protected void onPause() {
        super.onPause();
        manager.onPause();
    }

    @Override
    protected void onResume() {
        super.onResume();
        manager.onResume();
    }

    @Override
    protected void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);
        manager.onSaveInstanceState(outState);
    }
}

File: app/src/main/java/com/smoodi/smoodi/LoginActivity.kt
package com.smoodi.smoodi

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.Toast
import com.google.firebase.Firebase
import com.google.firebase.firestore.firestore
import com.google.mlkit.vision.codescanner.GmsBarcodeScanning
import com.smoodi.smoodi.data.Data

class LoginActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        val btScan: Button = findViewById(R.id.btScan)
        val btSuper: Button = findViewById(R.id.btSuper)

        btScan.setOnClickListener {

            val context = this@LoginActivity

            /* Scan Barcode */
            login("test")
//            GmsBarcodeScanning.getClient(context). also {
//                it.startScan()
//                    .addOnSuccessListener { barcode ->
//                        val rawValue: String? = barcode.rawValue
//                        login(rawValue!!)
//                    }
//                    .addOnFailureListener {
//                        val toast = getString(R.string.error_api)
//                        Toast.makeText(context, toast, Toast.LENGTH_SHORT).show()
//                    }
//            }
        }

        btSuper.setOnClickListener {
            val intent = Intent(this, SuperActivity::class.java)
            startActivity(intent)
        }
    }

    private fun login(id: String) {

        val db = Firebase.firestore

        db.collection(Data.KEY_USERS).document(id)
            .get()
            .addOnSuccessListener { document ->

                if (document.data == null) {

                    Data.id = id

                    val intent = Intent(this, JoinActivity::class.java)
                    startActivity(intent)

                } else {

                    val name = document.data!![Data.KEY_NAME].toString()
                    Data().setUserData(this, id, name)

                    val intent = Intent(this, MainActivity::class.java)
                    finishAffinity()
                    startActivity(intent)

                }

            }.addOnFailureListener {
                val toast = getString(R.string.error_server)
                Toast.makeText(this, toast, Toast.LENGTH_SHORT).show()
            }

    }
}